clients/sail_ui/lib/rpcs/thunder_rpc.dart

```diff
@@ -64,18 +67,152 @@ class ThunderLive extends ThunderRPC {
+      await _deleteWalletFiles(thunderDir);
+
+      // Try to set the seed multiple times
+      int setSeedRetries = 0;
+      const maxSetSeedRetries = 5;
+      const setSeedRetryDelay = Duration(milliseconds: 500);
+      Exception? lastError;
+
+      while (setSeedRetries < maxSetSeedRetries) {
+        try {
+          await _client().call('set_seed_from_mnemonic', [mnemonic]);
+          break;
+        } catch (e) {
+          lastError = Exception(e.toString());
+          setSeedRetries++;
+          if (setSeedRetries == maxSetSeedRetries) {
+            throw lastError;
+          }
+          await Future.delayed(setSeedRetryDelay);
+        }
+      }
+
+      // Verify the seed was set by checking if we can get a balance
+      int retries = 0;
+      const maxRetries = 20;
+      const retryDelay = Duration(seconds: 2);
+
+      while (retries < maxRetries) {
+        try {
+          await _client().call('balance');
+          return;
+        } catch (e) {
+          if (e.toString().contains("wallet doesn't have a seed")) {
+            retries++;
+            await Future.delayed(retryDelay);
+            continue;
+          }
+          rethrow;
+        }
+      }
+      throw Exception('Failed to verify wallet was ready after $maxRetries attempts');
+    } catch (e) {
+      log.e('Error setting Thunder seed', error: e);
+      rethrow;
+    }
+  }
+
+  Future<void> _deleteWalletFiles(String dataDir) async {
+    try {
+      // Delete wallet.mdb directory if it exists
+      final walletMdbDir = Directory(path.join(dataDir, 'wallet.mdb'));
+      if (await walletMdbDir.exists()) {
+        await walletMdbDir.delete(recursive: true);
+      }
+
+      // Delete data.mdb directory if it exists
+      final dataMdbDir = Directory(path.join(dataDir, 'data.mdb'));
+      if (await dataMdbDir.exists()) {
+        await dataMdbDir.delete(recursive: true);
+      }
+
+      // Delete wallet.dat if it exists
+      final walletDat = File(path.join(dataDir, 'wallet.dat'));
+      if (await walletDat.exists()) {
+        await walletDat.delete();
+      }
+
+      // Delete mnemonic.txt if it exists
+      final mnemonicTxt = File(path.join(dataDir, 'mnemonic.txt'));
+      if (await mnemonicTxt.exists()) {
+        await mnemonicTxt.delete();
+      }
+    } catch (e) {
+      log.e('Error deleting Thunder wallet files', error: e);
+      rethrow;
+    }
   }
 
   @override
   Future<void> stopRPC() async {
-    await _client().call('stop');
+    try {
+      await _client().call('stop');
+      // Wait a moment to let the node start shutting down
+      await Future.delayed(const Duration(milliseconds: 500));
+    } catch (e) {
+      // If we get a connection refused error, the node is likely already stopped
+      if (e.toString().contains('Connection refused') || e.toString().contains('Unknown Error')) {
+        return;
+      }
+      log.e('Error stopping RPC', error: e);
+      rethrow;
+    }
```

Comment:
I think this should only call `stop`, and nothing else. Like it was. awaiting, catching should be handled in the generic stop() function on the rpc_connection class.